### DISCUSSION TITLE
Two questions :: GDK_SCALE=2 steam && wayland

### DISCUSSION THREAD

**Discussion by opqriu** [netblue30/firejail#5848] [Q&A] [answered]
1.

I discovered Firejail, and ran it like this

firejail --private=$HOME/sandbox --seccomp-error-action=log steam

Is this command correct?
It actually worked very well.

But I run "~$ GDK_SCALE=2 steam" for HIDPI.

However, I can't seem to add the "GDK_SCALE=2" option in Firejail.

Failed1 :: firejail --private=$HOME/sandbox --seccomp-error-action=log steam GDK_SCALE=2

Failed2 :: firejail --private=$HOME/sandbox --seccomp-error-action=log GDK_SCALE=2 steam

Is there a good way to do it for me?

2.

I read a comment on reddit that firejail doesn't support wayland.

But i used firejail on wayland now (debian12, gnome43)

firefox and steam ran great (home isolation)

~$ firejail --private=$HOME/sandbox firefox

Is it pointless to use firejail on wayland?

Why do people say it's not supported?

**Comment by ghost** [comment, marked as answer]
> firejail --private=$HOME/sandbox --seccomp-error-action=log steam
Is this command correct?

That is a correct command. Firejail's desktop integration feature, controlled via the `firecfg` command, (amongst other stuff) creates symlinks under /usr/local/bin. So be aware of that possibility and always use `full paths` when running firejail from the command line (or in shell scripts). It avoids 'double sandboxes' and is considered 'good practice'.

> But I run "~$ GDK_SCALE=2 steam" for HIDPI.
However, I can't seem to add the "GDK_SCALE=2" option in Firejail.
[...]
Is there a good way to do it for me?

Sure thing. GDK_SCALE is just another environment variable and Firejail supports those.
Use --env=name=value on the command line or set it via ~/.config/firejail/steam.local:

```sh
$ cat ~/.config/firejail/steam.local
# Firejail profile for steam
# Persistent local customizations
env GDK_SCALE=2
```

> I read a comment on reddit that firejail doesn't support wayland.
[...]
Is it pointless to use firejail on wayland?

Firejail is as happy and able to be used on X11 as on Wayland. Like with all software there can be issues on both, but generally speaking Firejail performs very well on Wayland. You could [search our repo](https://github.com/search?q=repo%3Anetblue30%2Ffirejail+wayland&type=code), read around and form your own opinion(s) on this topic.

> Why do people say it's not supported?

Take your pick :-)

**Reply by opqriu** [reply]
This worked for me, 
thank you for your kind reply.

~$ firejail --private=$HOME/sandbox --seccomp-error-action=log --env=GDK_SCALE=2 steam

**Comment by rusty-snake** [comment]
> firejail --private=$HOME/sandbox --seccomp-error-action=log steam

As @glitsj16 already said, this is a correct (=valid) firejail command and using  the full path to steam would be better. Furthermore you should note that this sets the secccomp policy into auditing mode and does _not_ enforce it.

**Reply by opqriu** [reply]
I am happy to be able to use firejail correctly.  
Thanks to firejail, my experience with linux is more enjoyable. 
have a great day.
